### PR TITLE
[MLA] Update and sync MLA changes

### PR DESCRIPTION
**Pull request by m-liu**
# Description

Merge latest MLA optimizations
- 2026-05-26: Optimize new_kv_packing for larger kv_packing.
- 2026-05-21: Refactor internal logic of _pack_new_kv and add test for it.
- 2026-05-21: Optimize mask logic.
- 2026-05-15: Transpose KVC by XLUs for QK attention.
- 2026-05-15: Split bkv_sz into MXU size (256) for QK attention.
- 2026-05-15: Enable flexible KV packing sizes in MLA v2 KV fusion kernel.
- 2026-04-24: Add transpose_kv_cache option.
- 2026-04-23: Add one-step batched FlashAttention option to KV fusion kernel.

```
# 4 chip DSV3 results:
============ Serving Benchmark Result ============
Successful requests:                     896
Failed requests:                         0
Benchmark duration (s):                  619.57
Total input tokens:                      917504
Total generated tokens:                  7340032
Request throughput (req/s):              1.45
Output token throughput (tok/s):         11847.03
Peak output token throughput (tok/s):    15757.00
Peak concurrent requests:                896.00
Total token throughput (tok/s):          13327.90
```

### CODE REVIEW

**Review by gxd3 — APPROVED** [review]

**Review by gxd3 — APPROVED** [review]

File: tpu_inference/kernels/mla/v2/kernel.py [hunk 1]
```diff
@@ -24,35 +25,50 @@
 from jax.experimental.pallas import tpu as pltpu
 
 import tpu_inference.envs as envs
+from tpu_inference.kernels.mla.v2 import kv_utils
 from tpu_inference.kernels.mla.v2.transpose import xpose_pipeline
-from tpu_inference.logger import init_logger
 
-logger = init_logger(__name__)
+logger = logging.getLogger(__name__)
```
Comment (Xhark) [inline tpu_inference/kernels/mla/v2/kernel.py]: Any reason to change logger here?

**Review by Xhark — APPROVED** [review]

**Comment by m-liu** [comment]
perf updated in description after merging HEAD
